### ISSUE TITLE
Replace gcc with modern variant.

### ISSUE THREAD

**Issue by nterry** [deadcast2/UltraEd#84]
There is a modern toolchain for N64 located at https://github.com/tj90241/n64chain/releases/tag/9.1.0

You still need the includes, llibs and the nusys stuff from the official SDK, but you can delete the ultra/mipse directory, place the contents of the `bin`, `include`, `lib`, `libexec`, and `mips64-elf1` directories from the new toolchain in some location (i used gcc-9 in the ultra directory), and alter the steupgcc bat file to set the path to the new gcc bin directory.

For example, here's my new `setupgcc.bat` file:

setupgcc.bat (new)
> set gccdir=%ROOT%\gcc-9
>     path %gccdir%\bin;%path%
>     set gccsw=-mips3 -mgp32 -mfp32 -funsigned-char -D_LANGUAGE_C -D_ULTRA64 -D__EXTENSIONS__
>     set n64align=on
>     set GCC_CELF=ON
> 

I've tested it with a sample scene and it compiles just fine. Might be worth looking into.

What if you made swappable compilation backends. You could make it so the user could select which toolchain they wanted to use.

![new-gcc](https://user-images.githubusercontent.com/815254/102843612-c1445a80-43c6-11eb-9264-0264667adfdb.PNG)

(in the image above, Im also using the new makemask. See #83 for more details.

**Comment by deadcast2** [comment]
Hello! UltraEd is actually already using a newer version of GCC. I followed instructions from https://github.com/trhodeos/modern-n64sdk to build a new toolchain. That's how I was able to get the project finally running on something beyond Windows XP lol. I have seen that tool chain you mentioned and for some reason back then decided to go with what is currently being used. The swappable toolchain could be cool but not too sure what value it would bring unless I'm missing something?
